RoadTripMoustache/kana_to_kanji
# #100 Implement tile to display the kanji and vocabulary in the glossary

## 📖 Description


Changes:
- Implement `FuriganaText` to facilitate display of furigana (note that the `FuriganaText.vocabulary` factory constructor is temporary and will be removed/changed after the refactoring of the data structure)
- Implement `GlossaryListTile` to display the kanji and vocabulary of the respective tabs in the glossary
  - the `onPressed` is implemented but not visible until #102 is implemented
- Refactor the `KanjiList` and `VocabularyList` to:
  - make use of the new `GlossaryListTile`
  - keep the scroll position when changing tabs
  - reduce building time by using `ListView.builder`

### ⁉️ Related Issues




- #100 


### 🖼️ Screenshots:


<details>
    <summary>Demonstration</summary> 

 https://github.com/RoadTripMoustache/kana_to_kanji/assets/22211097/9678b8de-da9b-4b03-b836-e63ef69df480

</details>

### 🧪 How to test the change?




1. Open the app on the _Glossary_ page
2. Go to the _Kanji_ tab
3. See changes
4. Go to the _Vocabulary_ tab
5. See changes

### ☑️ Checklist before requesting a review
